### ISSUE TITLE
[feature] Fold non parallel sheet

### ISSUE THREAD

**Issue by Anthony-Gaudino** [shaise/FreeCAD_SheetMetal#295]
It seems that currently SheetMetal can only create bends on sheets with parallel surfaces.

Trying to create a bend where the bottom is not parallel results in a wrong bend:
<img width="274" alt="Screenshot 2024-01-17 at 20 15 58" src="https://github.com/shaise/FreeCAD_SheetMetal/assets/2439728/721b6ef0-36da-43ce-bebe-c00c3d33d1da">
[bad_bend.FCStd.zip](https://github.com/shaise/FreeCAD_SheetMetal/files/13968345/bad_bend.FCStd.zip)

If both top and bottom are not parallel it gives an error:
```
SheetMetalFoldCmd.py", line 266, in execute
    s = smFold(bendR = fp.radius.Value, bendA = fp.angle.Value, flipped = fp.invert, unfold = fp.unfold, kfactor = fp.kfactor, bendlinesketch = fp.BendLine,
  SheetMetalFoldCmd.py", line 122, in smFold
    toolFaces = tool.extrude(normal * -thk)
<class 'Part.OCCError'>: BRepSweep_Translation::Constructor
20:19:20  Recompute failed!
```

[broken_bend.FCStd.zip](https://github.com/shaise/FreeCAD_SheetMetal/files/13968417/broken_bend.FCStd.zip)

------

Having the possibility of bending non parallel surfaces would be very useful in some instances.

**Comment by shaise** [comment]
Yes. This a sheet metal limitation. It was meant to be used with flat sheets of metals.

**Comment by luzpaz** [comment]
> This a sheet metal limitation. It was meant to be used with flat sheets of metals.

@shaise are there plans to add this functionality eventually to SheetMetal WB ?

**Comment by shaise** [comment]
@luzpaz , I do not think this feature should be implemented. Its not in the scope of sheetmetal.

**Comment by luzpaz** [comment]
Theoretically, what (hypothetical) workbench would this be in the scope of?

**Comment by shaise** [comment]
Its a good question. This is a very specific workflow. Not sure what it is used for.

**Comment by luzpaz** [comment]
Based on the result of this discussion, I'll close this ticket (due to being out of scope).
